[BUG] DUE compilation error HAL_SPI.cpp non const spiDelayCyclesX4

### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

When compiling Marlin from both the 2.0.x and the bugfix-2-0-x branch the compilation fails with the following error:

<details><summary>Log Output</summary>

```
In file included from sketch/src/inc/MarlinConfig.h:49:0,
                 from sketch/src/MarlinCore.h:24,
                 from sketch/src/MarlinCore.cpp:31:
sketch/src/inc/SanityCheck.h:423:4: warning: #warning "MAX*_SS_PIN, MAX*_SS2_PIN, MAX*_CS_PIN, and MAX*_CS2_PIN are deprecated and will be removed in a future version. Please use TEMP_0_CS_PIN/TEMP_1_CS_PIN instead." [-Wcpp]
   #warning "MAX*_SS_PIN, MAX*_SS2_PIN, MAX*_CS_PIN, and MAX*_CS2_PIN are deprecated and will be removed in a future version. Please use TEMP_0_CS_PIN/TEMP_1_CS_PIN instead."
    ^
In file included from sketch/src/module/printcounter.h:25:0,
                 from sketch/src/MarlinCore.cpp:53:
sketch/src/module/../libs/duration_t.h:111:36: warning: unknown option after '#pragma GCC diagnostic' kind [-Wpragmas]
     #pragma GCC diagnostic ignored "-Wformat-overflow"
                                    ^
In file included from sketch/src/HAL/DUE/../../inc/MarlinConfig.h:49:0,
                 from sketch/src/HAL/DUE/HAL.cpp:27:
sketch/src/HAL/DUE/../../inc/SanityCheck.h:423:4: warning: #warning "MAX*_SS_PIN, MAX*_SS2_PIN, MAX*_CS_PIN, and MAX*_CS2_PIN are deprecated and will be removed in a future version. Please use TEMP_0_CS_PIN/TEMP_1_CS_PIN instead." [-Wcpp]
   #warning "MAX*_SS_PIN, MAX*_SS2_PIN, MAX*_CS_PIN, and MAX*_CS2_PIN are deprecated and will be removed in a future version. Please use TEMP_0_CS_PIN/TEMP_1_CS_PIN instead."
    ^
In file included from sketch/src/HAL/DUE/../../inc/MarlinConfig.h:49:0,
                 from sketch/src/HAL/DUE/HAL_SPI.cpp:40:
sketch/src/HAL/DUE/../../inc/SanityCheck.h:423:4: warning: #warning "MAX*_SS_PIN, MAX*_SS2_PIN, MAX*_CS_PIN, and MAX*_CS2_PIN are deprecated and will be removed in a future version. Please use TEMP_0_CS_PIN/TEMP_1_CS_PIN instead." [-Wcpp]
   #warning "MAX*_SS_PIN, MAX*_SS2_PIN, MAX*_CS_PIN, and MAX*_CS2_PIN are deprecated and will be removed in a future version. Please use TEMP_0_CS_PIN/TEMP_1_CS_PIN instead."
    ^
In file included from sketch/src/HAL/DUE/HAL_SPI.cpp:41:0:
sketch/src/HAL/DUE/HAL_SPI.cpp: In function 'uint8_t spiTransferX(uint8_t)':
sketch/src/HAL/DUE/../shared/Delay.h:93:83: error: the value of 'spiDelayCyclesX4' is not usable in a constant expression
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
sketch/src/HAL/DUE/HAL_SPI.cpp:252:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/HAL_SPI.cpp:243:19: note: 'uint32_t spiDelayCyclesX4' is not const
   static uint32_t spiDelayCyclesX4 = 4 * (F_CPU) / 1000000; // 4µs => 125khz
                   ^
In file included from sketch/src/HAL/DUE/HAL_SPI.cpp:41:0:
sketch/src/HAL/DUE/../shared/Delay.h:93:83: note: in template argument for type 'bool' 
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
sketch/src/HAL/DUE/HAL_SPI.cpp:252:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/../shared/Delay.h:93:83: error: the value of 'spiDelayCyclesX4' is not usable in a constant expression
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
sketch/src/HAL/DUE/HAL_SPI.cpp:252:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/HAL_SPI.cpp:243:19: note: 'uint32_t spiDelayCyclesX4' is not const
   static uint32_t spiDelayCyclesX4 = 4 * (F_CPU) / 1000000; // 4µs => 125khz
                   ^
In file included from sketch/src/HAL/DUE/HAL_SPI.cpp:41:0:
sketch/src/HAL/DUE/../shared/Delay.h:93:83: note: in template argument for type 'int' 
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
sketch/src/HAL/DUE/HAL_SPI.cpp:252:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/../shared/Delay.h:93:95: error: invalid type in declaration before '(' token
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                               ^
sketch/src/HAL/DUE/HAL_SPI.cpp:252:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/../shared/Delay.h:93:83: error: the value of 'spiDelayCyclesX4' is not usable in a constant expression
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
sketch/src/HAL/DUE/HAL_SPI.cpp:257:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/HAL_SPI.cpp:243:19: note: 'uint32_t spiDelayCyclesX4' is not const
   static uint32_t spiDelayCyclesX4 = 4 * (F_CPU) / 1000000; // 4µs => 125khz
                   ^
In file included from sketch/src/HAL/DUE/HAL_SPI.cpp:41:0:
sketch/src/HAL/DUE/../shared/Delay.h:93:83: note: in template argument for type 'bool' 
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
sketch/src/HAL/DUE/HAL_SPI.cpp:257:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/../shared/Delay.h:93:83: error: the value of 'spiDelayCyclesX4' is not usable in a constant expression
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
sketch/src/HAL/DUE/HAL_SPI.cpp:257:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/HAL_SPI.cpp:243:19: note: 'uint32_t spiDelayCyclesX4' is not const
   static uint32_t spiDelayCyclesX4 = 4 * (F_CPU) / 1000000; // 4µs => 125khz
                   ^
In file included from sketch/src/HAL/DUE/HAL_SPI.cpp:41:0:
sketch/src/HAL/DUE/../shared/Delay.h:93:83: note: in template argument for type 'int' 
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
sketch/src/HAL/DUE/HAL_SPI.cpp:257:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
sketch/src/HAL/DUE/../shared/Delay.h:93:95: error: invalid type in declaration before '(' token
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                               ^
sketch/src/HAL/DUE/HAL_SPI.cpp:257:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
exit status 1
Fehler beim Kompilieren für das Board Arduino Due (Programming Port).
```
</details>

After digging into the code I found out that the problem is the usage of the MACRO 
#define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
in the file Delay.h in conjunction with the bad definition of 
static uint32_t spiDelayCyclesX4 = 4 * (F_CPU) / 1000000; // 4µs => 125khz
in HAL_SPI.cpp

Changing the variable definition to static const uint32_t spiDelayCyclesX4 = 4 * (F_CPU) / 1000000; // 4µs => 125khz does solve above error but gives a new one because spiDelayCyclesX4 is tryed to by overridden in void spiInit(uint8_t spiRate) in the default case of switch/case instuction.
Commenting this line out 
spiDelayCyclesX4 = ((F_CPU) / 1000000) >> (6 - spiRate) << 2; // spiRate of 2 gives the maximum error with current CPU

makes the build finally go through successfully.

### Bug Timeline

New

### Expected behavior

Build shall work flawless for Arduino DUE board

### Actual behavior

Build is breaking, see description above

### Steps to Reproduce

1. Select Arduino DUE board in the board-configurator
2. Start build

### Version of Marlin Firmware

2.0.x, bugfix-2.0.x

### Printer model

CoreXY

### Electronics

Arduino DUE with RAMPS

### Add-ons

_No response_

### Your Slicer

Simplify3D

### Host Software

SD Card (headless)

### Additional information & file uploads

_No response_

using arduino ide 1.8.13 does produce same error (without all the extra warnings)

```
In file included from /tmp/arduino_build_429719/sketch/src/HAL/DUE/HAL_SPI.cpp:41:0:
/tmp/arduino_build_429719/sketch/src/HAL/DUE/HAL_SPI.cpp: In function 'uint8_t spiTransferX(uint8_t)':
/tmp/arduino_build_429719/sketch/src/HAL/DUE/../shared/Delay.h:93:83: error: the value of 'spiDelayCyclesX4' is not usable in a constant expression
   #define DELAY_CYCLES(X) do { SmartDelay<IS_CONSTEXPR(X), IS_CONSTEXPR(X) ? X : 0> _smrtdly_X(X); } while(0)
                                                                                   ^
/tmp/arduino_build_429719/sketch/src/HAL/DUE/HAL_SPI.cpp:252:7: note: in expansion of macro 'DELAY_CYCLES'
       DELAY_CYCLES(spiDelayCyclesX4);
       ^
/tmp/arduino_build_429719/sketch/src/HAL/DUE/HAL_SPI.cpp:243:19: note: 'uint32_t spiDelayCyclesX4' is not const
   static uint32_t spiDelayCyclesX4 = 4 * (F_CPU) / 1000000; // 4µs => 125khz
                   ^

```

--- a/Marlin/src/HAL/DUE/HAL_SPI.cpp
+++ b/Marlin/src/HAL/DUE/HAL_SPI.cpp
@@ -240,7 +240,7 @@
   }
 
   // all the others
-  static uint32_t spiDelayCyclesX4 = 4 * (F_CPU) / 1000000; // 4µs => 125khz
+  static uint16_t spiDelayNS = 400; // 400ns => 125khz
 
   static uint8_t spiTransferX(uint8_t b) { // using Mode 0
     int bits = 8;
@@ -249,12 +249,12 @@
       b <<= 1; // little setup time
 
       WRITE(SD_SCK_PIN, HIGH);
-      DELAY_CYCLES(spiDelayCyclesX4);
+      DELAY_NS(spiDelayNS);
 
       b |= (READ(SD_MISO_PIN) != 0);
 
       WRITE(SD_SCK_PIN, LOW);
-      DELAY_CYCLES(spiDelayCyclesX4);
+      DELAY_NS(spiDelayNS);
     } while (--bits);
     return b;
   }
@@ -510,7 +510,7 @@
         spiRxBlock = (pfnSpiRxBlock)spiRxBlockX;
         break;
       default:
-        spiDelayCyclesX4 = ((F_CPU) / 1000000) >> (6 - spiRate) << 2; // spiRate of 2 gives the maximum error with current CPU
+        spiDelayNS = (((F_CPU) / 1000000) >> (6 - spiRate) << 2) * 100; // spiRate of 2 gives the maximum error with current CPU
         spiTransferTx = (pfnSpiTransfer)spiTransferX;
         spiTransferRx = (pfnSpiTransfer)spiTransferX;
         spiTxBlock = (pfnSpiTxBlock)spiTxBlockX;

--- a/Marlin/src/HAL/DUE/HAL_SPI.cpp
+++ b/Marlin/src/HAL/DUE/HAL_SPI.cpp
@@ -240,7 +240,7 @@
   }
 
   // all the others
-  static uint16_t spiDelayNS = 400; // 400ns => 125khz
+  static uint16_t spiDelayNS = 4000; // 400ns => 125khz
 
   static uint8_t spiTransferX(uint8_t b) { // using Mode 0
     int bits = 8;
@@ -510,7 +510,7 @@
         spiRxBlock = (pfnSpiRxBlock)spiRxBlockX;
         break;
       default:
-        spiDelayNS = (((F_CPU) / 1000000) >> (6 - spiRate) << 2) * 100; // spiRate of 2 gives the maximum error with current CPU
+        spiDelayNS = 4000 >> (6 - spiRate); // spiRate of 2 gives the maximum error with current CPU
         spiTransferTx = (pfnSpiTransfer)spiTransferX;
         spiTransferRx = (pfnSpiTransfer)spiTransferX;
         spiTxBlock = (pfnSpiTxBlock)spiTxBlockX;

--- a/Marlin/src/HAL/DUE/HAL_SPI.cpp
+++ b/Marlin/src/HAL/DUE/HAL_SPI.cpp
@@ -240,7 +240,7 @@
   }
 
   // all the others
-  static uint16_t spiDelayNS = 4000; // 400ns => 125khz
+  static uint16_t spiDelayNS = 4000; // 4000ns => 125khz
 
   static uint8_t spiTransferX(uint8_t b) { // using Mode 0
     int bits = 8;

